### DISCUSSION TITLE
WebWorker cache accumulation issue in OffscreenCanvas example

### DISCUSSION THREAD

**Discussion by eternityK1** [openlayers/openlayers#16273] [Q&A] [answered]
Hello OpenLayers team,

I’ve encountered an issue in the OffscreenCanvas example where the WebWorker accumulates a significant amount of cache over time without clearing it, leading to excessive memory usage. In my case, the Web Worker reached 1.5 GB of memory usage.

Steps to reproduce:

1. Open the OffscreenCanvas example. https://openlayers.org/en/v10.2.1/examples/offscreen-canvas.html
2. Begin navigating around the map, especially in areas with a high density of geometry, such as cities.
3. Move between multiple cities with complex geometries.
4. Over time, observe how memory usage grows in the DevTools

Thank you for your excellent work on OpenLayers! I hope this feedback proves helpful. I’m happy to provide further details if necessary.

Environment:
 • Browser: Google Chrome 129.0.6668.90, Firefox 131.0.2 
 • OpenLayers version: 10.2.1

![offScreen canvas](https://github.com/user-attachments/assets/8b436db5-9346-4965-8802-c43f47546129)

**Comment by ahocevar** [comment, marked as answer]
This is a bug. See #16276.

**Comment by eternityK1** [comment]
Thanks for your solution. @ahocevar 

I noticed that on 4k screens in full screen mode in the firefox browser, the solution with offscreencanvas and ol-mapbox-style in WebWorker works slower than rendering in the main thread. In chromium browsers, on the contrary, there is a performance increase.

I wonder what is the reason for this situation?

**Comment by ahocevar** [comment]
Browser implementation details - I could also only guess.
